agent/agent.go

```diff
@@ -32,6 +32,8 @@ type Config struct {
 	TTL             time.Duration
 }
 
+func (a *Agent) DoCheck() (CheckResult, error) { return a.check() }
```

Comment:
Hmm, I don't quite like this to be honest, it feels like some avoidable repetition.

It would make sense if we offloaded more work to this function by completely encapsulating the `Checker` here, which means we don't want to leak its internal details like the `CheckResult` struct; so we could then probably rename this method to `RunChecks`, and put the output logic invocation here, maybe by passing a `Writer` interface so that we're not hard coding side effects on the stdout.

Either this, or just make `check` public and call it from the command handler as it is now. ;)

In general, if you have a `SomeVerb` method, I find having another `DoSomeVerb` method somewhere else to be a little of an anti-pattern. You are either doing that thing, or doing something else! ;)

Reply:
Fixed. I have moved the call inside the agent/agent/go, there is no more need to export the method.